tests/fuzz/fuzz-open_channel.c

```diff
@@ -0,0 +1,452 @@
+			sign_hash(&p, &h, &sig.s);
+			sig.sighash_type = SIGHASH_ALL;
+
+			return towire_hsmd_sign_tx_reply(ctx, &sig);
+		}
+		else
+			assert(false && "Too many HSMD reads!");
+	}
+	else if (fd == PEER_FD) /* Peer message */
+	{
+		/* Choose between creating a valid message and a fuzzer one. */
+		if (fromwire_u8(cursor, max) % 2) {
+			struct privkey p;
+			struct sha256_double h;
+			struct bitcoin_signature sig;
+
+			memset(&h, 1, sizeof(h));
+			memset(&p, 2, sizeof(p));
+			sign_hash(&p, &h, &sig.s);
+			fromwire_bitcoin_outpoint(cursor, max, &state->funding);
+
+			return towire_funding_created(ctx, &state->channel_id,
+					&state->funding.txid, state->funding.n, &sig.s);
+		}
+		else
+			return create_fuzz_msg(ctx);
+	}
+	else
+		assert(false && "Unknown caller!");
+}
+
+static bool test_sync_write(int fd, const void *msg TAKES)
+{
+	if (fd == REQ_FD) /* lightningd message */
+	{
+		struct amount_sat funding_satoshis;
+		struct amount_msat push_msat;
+		struct amount_sat dust_limit_satoshis;
+		struct amount_msat max_htlc_value_in_flight_msat;
+		struct amount_sat channel_reserve_satoshis;
+		struct amount_msat htlc_minimum_msat;
+		u32 feerate_per_kw;
+		u16 to_self_delay;
+		u16 max_accepted_htlcs;
+		u8 channel_flags;
+		u8 *shutdown_scriptpubkey;
+
+		assert(fromwire_openingd_got_offer(tmpctx, msg,
+					&funding_satoshis,
+					&push_msat,
+					&dust_limit_satoshis,
+					&max_htlc_value_in_flight_msat,
+					&channel_reserve_satoshis,
+					&htlc_minimum_msat,
+					&feerate_per_kw,
+					&to_self_delay,
+					&max_accepted_htlcs,
+					&channel_flags,
+					&shutdown_scriptpubkey));
+	}
+	else if (fd == HSM_FD) /* HSMD message */
+	{
+		if (hsmd_writes++ == 0) {
+			bool is_outbound;
+			struct amount_sat channel_value;
+			struct amount_msat push_value;
+			struct bitcoin_txid funding_txid;
+			u16 funding_txout, local_to_self_delay, remote_to_self_delay;
+			u8 *local_shutdown_script, *remote_shutdown_script;
+			u32 *local_shutdown_wallet_index;
+			struct basepoints remote_basepoints;
+			struct pubkey remote_funding_pubkey;
+			struct channel_type *channel_type;
+
+			assert(fromwire_hsmd_setup_channel(tmpctx, msg,
+						&is_outbound,
+						&channel_value,
+						&push_value,
+						&funding_txid,
+						&funding_txout,
+						&local_to_self_delay,
+						&local_shutdown_script,
+						&local_shutdown_wallet_index,
+						&remote_basepoints,
+						&remote_funding_pubkey,
+						&remote_to_self_delay,
+						&remote_shutdown_script,
+						&channel_type));
+
+		} else if (hsmd_writes++ == 1) {
+			struct bitcoin_tx *tx;
+			struct pubkey remote_funding_key, remote_per_commit;
+			bool option_static_remotekey;
+			u64 commit_num;
+			struct hsm_htlc *htlcs;
+			u32 feerate;
+
+			assert(fromwire_hsmd_sign_remote_commitment_tx(tmpctx, msg,
+								&tx,
+								&remote_funding_key,
+								&remote_per_commit,
+								&option_static_remotekey,
+								&commit_num,
+								&htlcs,
+								&feerate));
+		}
+		else
```

Comment:
We also need to handle HSM messages for signature validation in `validate_initial_commitment_signature`.